network/p2p/fixture_test.go

```diff
@@ -136,16 +143,15 @@ func nodeFixture(
 	err = n.WithDefaultUnicastProtocol(parameters.handlerFunc, parameters.unicasts)
 	require.NoError(t, err)
 
-	require.Eventuallyf(t, func() bool {
-		ip, p, err := n.GetIPPort()
-		return err == nil && ip != "" && p != ""
-	}, 3*time.Second, ticksForAssertEventually, fmt.Sprintf("could not start node %s", identity.NodeID.String()))
+	//require.Eventuallyf(t, func() bool {
```

Comment:
I updated the topic validator tests to use **_nodeFixture_** instead of **_createNode_** , for some reason this call **_require.Eventuallyf_** adds flakiness to the test. It is causing the [context](https://github.com/onflow/flow-go/blob/khalil/1704-masquerade-slashing-hooks/network/p2p/topic_validator_test.go#L91) to time out in the happy path of the topic validator test. For some reason the messages are published but not delivered to other libP2P nodes. When I comment out this call to **_require.Eventuallyf_** the flakiness goes away. 

cc: @yhassanzadeh13

Reply:
The short answer is to set the tick value to `10ms` (instead of current `100ms`). The `Eventuallyf` call is blocking for the first tick, i.e., the first invocation of the injected function happens with a delay of the passed tick value. So a smaller tick value makes it blocking shorter. 

Moreover, it seems the injected function to `Eventuallyf` does not matter. I tried replacing the current one with just returning true, and it fails for the tick of `100ms` and is passing for `10ms`. 

Having 3 nodes in this test, it concludes that a delay of `300ms` leads it to failure, but a delay of `30ms` is fine. 

Despite the test is passing with `10ms` the failure reason for `100ms` is still vague. It is evident that the data flow in `TestAuthorizedSenderValidator_Unauthorized` needs improvement. For example, the same context is initialized in multiple places (checkout `timedCtx`), which makes debugging the tests convoluted. 

I checked and it seems the test is originally developed by @smnzhu, so I urge him please to improve its code quality and investigate this issue so that it won't later on appear as a flakey test.